### PR TITLE
SCUMM: Initialize class members

### PR DESCRIPTION
**Pull request by bluegr**
After @eriktorbjorn's work on SCUMM class member initialization in PR #6233, I went ahead and used ReSharper to initialize all class members. Some of these initializers are superfluous, as the variables are then initialized in `init()` methods, but there's no harm done in initializing them anyway.

This includes @eriktorbjorn's commits from PR #6233, so this is a superset of that PR.

### CODE REVIEW

File: engines/scumm/he/gfx_comp/aux_comp.cpp
Comment (AndywinXp) [inline engines/scumm/he/gfx_comp/aux_comp.cpp:1]: What's happening here? Why are these pointers removed? 🤔

File: engines/scumm/he/logic/funshop.cpp [hunk 1]
```diff
@@ -67,7 +67,7 @@ int32 LogicHEfunshop::dispatch(int op, int numArgs, int32 *args) {
 void LogicHEfunshop::op_1004(int32 *args) {
 	double data[8], at, sq;
 	int32 x, y;
-	int i=0;
+	int i;
```
Comment (AndywinXp) [inline engines/scumm/he/logic/funshop.cpp:70]: `i = 0` instead of `i=0`

File: engines/scumm/he/moonbase/moonbase.h [hunk 2]
```diff
@@ -42,9 +42,6 @@ class Moonbase {
 	void deallocateArray(int array);
 	int callScummFunction(int scriptNumber, int paramCount,...);
 
-	void blitDistortion(byte *bufferData, const int bufferWidth, const int bufferHeight, const int bufferPitch,
```
Comment (AndywinXp) [inline engines/scumm/he/moonbase/moonbase.h:45]: Why? 🤔

File: engines/scumm/scumm_v0.h [hunk 3]
```diff
@@ -152,7 +152,6 @@ class ScummEngine_v0 : public ScummEngine_v2 {
 	void o_lights();
 	void o_unlockCostume();
 	void o_unlockScript();
-	void o_decrement();
```
Comment (AndywinXp) [inline engines/scumm/scumm_v0.h:155]: Why?

**Review by AndywinXp — CHANGES_REQUESTED** [review]
Great work overall, thank you for taking the time to address all of these things! I have a few comments, mostly asking for the reason of certain changes.

File: engines/scumm/he/moonbase/moonbase.h (hunk 2 above)
Comment (bluegr) [reply engines/scumm/he/moonbase/moonbase.h:45]: It's not implemented

File: engines/scumm/scumm_v0.h (hunk 3 above)
Comment (bluegr) [reply engines/scumm/scumm_v0.h:155]: It's not implemented

File: engines/scumm/he/logic/funshop.cpp (hunk 1 above)
Comment (bluegr) [reply engines/scumm/he/logic/funshop.cpp:70]: This gets reassigned immediately, so there's no reason for that assignment

File: engines/scumm/he/gfx_comp/aux_comp.cpp
Comment (bluegr) [reply engines/scumm/he/gfx_comp/aux_comp.cpp:1]: They're not used from any code after the initial assignment

Comment (AndywinXp) [reply engines/scumm/he/gfx_comp/aux_comp.cpp:1]: Cool, thank you!

File: engines/scumm/he/logic/funshop.cpp (hunk 1 above)
Comment (AndywinXp) [reply engines/scumm/he/logic/funshop.cpp:70]: ...Sorry, I read green and red the opposite way. I swear I'm not daltonic 😛

File: engines/scumm/he/moonbase/moonbase.h (hunk 2 above)
Comment (AndywinXp) [reply engines/scumm/he/moonbase/moonbase.h:45]: Right, sorry, I forgot I moved this declaration to `wiz_he.h`

**Review by AndywinXp — APPROVED** [review]

**Comment by bluegr** [comment]
Great :) Thanks for the review

**Pull request merged**